src/idiomatic/leveraging-the-type-system/raii.md

```diff
@@ -0,0 +1,116 @@
+---
+minutes: 30
+---
+
+# RAII and `Drop` in Practice
+
+RAII (_Resource Acquisition Is Initialization_) means tying the lifetime of a
+resource to the lifetime of a value.
+
+Rust applies RAII automatically for memory management. The `Drop` trait lets you
+extend this pattern to anything else.
+
+```rust
+use std::sync::Mutex;
+
+fn main() {
+    let mux = Mutex::new(vec![1, 2, 3]);
+
+    {
+        let mut data = mux.lock().unwrap();
+        data.push(4); // lock held here
+    } // lock automatically released here
+}
+```
+
+<details>
+
+- In the above example
+  [the `Mutex`](https://doc.rust-lang.org/std/sync/struct.Mutex.html) owns its
+  data: you can’t access the value inside without first acquiring the lock.
+
+  `mux.lock()` returns a
+  [`MutexGuard`](https://doc.rust-lang.org/std/sync/struct.MutexGuard.html),
+  which [dereferences](https://doc.rust-lang.org/std/ops/trait.DerefMut.html) to
+  the data and implements
+  [`Drop`](https://doc.rust-lang.org/std/ops/trait.Drop.html).
+
+- You may recall from
+  [the Memory Management chapter](../../memory-management/drop.md) that the
+  [`Drop` trait](https://doc.rust-lang.org/std/ops/trait.Drop.html) lets you
+  define what should happen when a resource is dropped.
+
+  - In
```

Comment:
It's probably only necessary to include one "callback" to Fundamentals -- the important point is to that this slide is a quick review of previous content, and if students need a deeper refresher they can find that content in the Fundamentals course.

That said, these speaker notes are pretty long! Is it possible to trim this down to just call out the bits necessary for the RAII patterns introduced here, leaving the rest to the students' memory of Fundamentals?